depthai_nodes/ml/parsers/parser.py

```diff
@@ -13,17 +27,15 @@ def __init__(self):
 
         self.head_config: Dict = {}
 
-    def build(self, heads: Union[List, Dict], head_name: str = ""):
-        """Initial build method for all parsers. The method sets the head configuration
-        for the parser.
+    def build(self, heads: Union[List, Dict], head_name: str = None):
```

Comment:
After all the discussion I think its best to change this to
```
def build(self, nn_archive: depthai.NNArchive, head_names: List[str] = None):
```
And add logic to get the metadata from nn_archive + instantiate parsers based on parser name and head name from the metadata.